resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -498,7 +498,7 @@ protected void initChannel(DatagramChannel ch) {
         } else {
             future = b.bind(localAddress);
         }
-        Throwable cause = future.cause();
+        Throwable cause = future.awaitUninterruptibly().cause();
```

Comment:
ok to call a blocking method here?

Reply:
Same concern. Should we use timeout instead?

Reply:
`DnsNameResolvers` are usually not built in event loop threads, no? And constructor can't return a future. If there's a timeout, what should it be by default?

Reply:
1000ms? or 1500ms?

Reply:
> `DnsNameResolvers` are usually not built in event loop threads, no? And constructor can't return a future. If there's a timeout, what should it be by default?

There is a use case: when you what to create a remote call.

client -> server ( + client for remote calls) -> backend

Then the init, register and connect channel will happen in the server's event loop, in case you do not offload the remote call in another thread.

This tries to obtain a resolver on resolve and connect step
https://github.com/netty/netty/blob/d99a8f75b4cecfc6823cc8ec4d305958a7b3c90d/transport/src/main/java/io/netty/bootstrap/Bootstrap.java#L194

A new resolver is created if it is needed
https://github.com/netty/netty/blob/d99a8f75b4cecfc6823cc8ec4d305958a7b3c90d/resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java#L70

This one creates a new DnsNameResolver
https://github.com/netty/netty/blob/d99a8f75b4cecfc6823cc8ec4d305958a7b3c90d/resolver-dns/src/main/java/io/netty/resolver/dns/DnsAddressResolverGroup.java#L92